### PR TITLE
Feature/nirmal agnihotri/1368492449 agency table link

### PR DESCRIPTION
**Pull request by NirmalAgnihotri**
### Administrative Info

Monday Board ID: 851212620
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?
- Entire table row is clickable
- Dates changed/removed during agency editing have respective notes removed from DB
- UserSelected dates can be deleted again after update that removed -7 from DB
- Handled edge case: Removing a distribution days at the same time as a retail day will no longer remove both

### Testing

How did you confirm your changes worked?

- Added notes for days and removed days while editing (through calendar and checkbox) and checked for note removal
- deleted a userselected day and checked for change

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
Add a note for a day. Remove that day by unchecking its pattern or deselecting it in the calendar while editing an agency. Check the database to see that the note is removed. This will work for both retail and distribution.
Add a retail and distribution time at the same time. Remove one and check that the other is still there.

### CODE REVIEW

File: frontend/src/components/Calendar/NotesModal.js [hunk 1]
```diff
@@ -80,8 +80,10 @@ function NotesModal({ showModal, toggleModal, selectedEvent, deleted, changeDele
           .then((res) => res.json())
           .then((data) => data.agency)
           .then((agency) => {
-            const regex = /:[0-9][0-9]-/g; // regex to remove seconds from standard ISO format
-            const deleteDate = selectedEvent.event.startStr.replace(regex, "-");
+            const eventTime = selectedEvent.event.startStr;
+            const endIndex = eventTime.indexOf(":", eventTime.indexOf(":") + 1);
+            const deleteDate = eventTime.substring(0, endIndex);
+            console.log(deleteDate);
```
Comment (sblumin) [inline frontend/src/components/Calendar/NotesModal.js:86]: is this console.log necessary?

File: frontend/src/components/Calendar/NotesModal.js [hunk 2]
```diff
@@ -234,6 +249,7 @@ function NotesModal({ showModal, toggleModal, selectedEvent, deleted, changeDele
 
   useEffect(() => {
     if (selectedEvent) {
+      console.log(selectedEvent.event.startStr);
```
Comment (sblumin) [inline frontend/src/components/Calendar/NotesModal.js:252]: same with this one?

**Review by sblumin — COMMENTED** [review]
Left two comments on console.log lines but otherwise looks great!!! Please address whether these are needed or not and then we can merge!

**Review by StephenTan12 — COMMENTED** [review]
Besides Sara's comments, looks good guys!

File: frontend/src/components/Calendar/NotesModal.js (hunk 2 above)
Comment (NirmalAgnihotri) [reply frontend/src/components/Calendar/NotesModal.js]: note necessary, removed.

**Review by sblumin — APPROVED** [review]
LGTM!! Great work y'all and thank you for getting this in on such a tight schedule, it's much appreciated!

**Review by StephenTan12 — APPROVED** [review]
Lgtm

**Pull request merged**